.github/workflows/build_commits.yaml

```diff
@@ -0,0 +1,56 @@
+name: build_commits
+
+on: [pull_request]
+
+jobs:
+  build_commits:
+    name: Check if build works for every commit
+    runs-on: ubuntu-latest
+    steps:
+      - name: Configure git
+        run: |
+          git config --global user.name "GitHub Actions"
+          git config --global user.email "github-actions@users.noreply.github.com"
+
+      - name: Install Go
+        uses: actions/setup-go@37335c7bb261b353407cff977110895fa0b4f7d8
+        with:
+          go-version: 1.16.3
+          
+      - name: Cache LLVM and Clang
+        id: cache-llvm
+        uses: actions/cache@26968a09c0ea4f3e233fdddbafd1166051a095f6
+        with:
+          path: $HOME/.clang
+          key: llvm-10.0
+          
+      - name: Install LLVM and Clang
+        uses: KyleMayes/install-llvm-action@4609a7b44dbb15353bdb46dafaac2914948399c9
+        with:
+          version: "10.0"
+          directory: $HOME/.clang
+          cached: ${{ steps.cache-llvm.outputs.cache-hit }}
+          
+      - name: Install libelf
+        run: |
+          sudo apt-get install libelf-dev
+          
+      - name: Checkout code
+        uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+        with:
+          fetch-depth: 0
```

Comment:
```suggestion
          ref: ${{ github.event.pull_request.head.sha }}
          fetch-depth: 0
```
Since apparently `GITHUB_SHA` defaults to a "virtual merge SHA" of the PR into target branch:
```
Run actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
Syncing repository: cilium/cilium
Getting Git version info
Deleting the contents of '/home/runner/work/cilium/cilium'
Initializing the repository
Disabling automatic garbage collection
Setting up auth
Fetching the repository
Determining the checkout info
Checking out the ref
  /usr/bin/git checkout --progress --force refs/remotes/pull/15659/merge
  Note: switching to 'refs/remotes/pull/15659/merge'.
  You are in 'detached HEAD' state. You can look around, make experimental
  changes and commit them, and you can discard any commits you make in this
  state without impacting any branches by switching back to a branch.
  If you want to create a new branch to retain commits you create, you may
  do so (now or later) by using -c with the switch command. Example:
    git switch -c <new-branch-name>
  Or undo this operation with:
    git switch -
  Turn off this advice by setting config variable advice.detachedHead to false
  HEAD is now at b73c40215 Merge 630d39832fc1fe760a1b6db45be7f83f1e06be76 into 99eff1b46c37a9a47d31cd21c09230110c5c3c52
/usr/bin/git log -1 --format='%H'
'b73c402151f902d53db1b4cf7f897ecf9472d928
```